copter: fix condition yaw early completion

IamPete1 opened the pull request:
This fixes a long standing condition yaw bug. The verify function checks if current yaw equals target yaw, however that target yaw is still being slewed towards the final desired yaw. Currently condition yaws almost always instantly complete. This is a change in behavior in some cases, depending what comes after the condition yaw.

rmackay9 commented on ArduCopter/mode_auto.cpp:
```diff
@@ -1871,13 +1871,11 @@ bool ModeAuto::verify_within_distance()
-        auto_yaw.set_mode(AUTO_YAW_FIXED);
-    }
+    // make sure still in fixed yaw mode, the waypoint controller often retakes control of yaw as it executes a new waypoint command
+    auto_yaw.set_mode(AUTO_YAW_FIXED);
```
In general I'm not a fan of us doing any state changes in the verify_ methods.  Instead we should put the state changes into the run.  Still, it's an existing issue so we can push that off to another day.

rmackay9 reviewed (approved):
You've tested this of course?

By the way, I suspect, like Rover, we could do a better job of slewing the yaw using an acceleration limit as well but again.. not today.

IamPete1 commented:
@rmackay9 Yes I have tested, I added a little send text on condition yaw completion, I have deleted the logs now tho.... I can re-run the test if you like.

rmackay9 commented:
Merged, thanks!
